tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/contact/OpenPaasContactsConsumer.java

```diff
@@ -117,13 +127,18 @@ private Mono<EmailAddressContact> messageConsume(AcknowledgableDelivery ackDeliv
             });
     }
 
-    private Mono<EmailAddressContact> handleMessage(ContactAddedRabbitMqMessage contactAddedMessage) {
-        LOGGER.trace("Consumed jCard object message: {}", contactAddedMessage);
-        Optional<ContactFields> maybeContactFields = contactAddedMessage.vcard().asContactFields();
+    private Mono<Object> handleMessage(ContactRabbitMqMessage contactMessage, String queue) {
+        LOGGER.trace("Consumed jCard object message: {}", contactMessage);
+        Optional<ContactFields> maybeContactFields = contactMessage.vcard().asContactFields();
         return maybeContactFields.map(
-                openPaasContact -> openPaasRestClient.retrieveMailAddress(contactAddedMessage.userId())
+                openPaasContact -> openPaasRestClient.retrieveMailAddress(contactMessage.userId())
                     .map(this::getAccountIdFromMailAddress)
-                    .flatMap(ownerAccountId -> indexContactIfNeeded(ownerAccountId, openPaasContact)))
+                    .flatMap(ownerAccountId -> switch (queue) {
+                        case QUEUE_NAME_ADD -> indexContactIfNeeded(ownerAccountId, openPaasContact);
+                        case QUEUE_NAME_UPDATE -> updateContact(ownerAccountId, openPaasContact);
+                        case QUEUE_NAME_DELETE -> deleteContact(ownerAccountId, openPaasContact);
+                        default -> Mono.empty();
+                    }))
```

Comment:
That's uggly, i'd prefer we have indirection with a lambda: in the consume, we declare how to consume a message for each queue and pass it down the stack and get rid of that tricky tricky switch.

BTW default option should have been `throw new Exception("Oh my god UNSUPPORTED");`